### ISSUE TITLE
Standardized Belgif problem type(s) for 409 Conflict

### ISSUE THREAD

**Issue by jpraet** [belgif/rest-guide#185]
The Belgif REST Guidelines make various references to HTTP 409 Conflict:

* optimistic locking in [PUT](https://www.belgif.be/specification/rest/api-guide/#document-full-update) / [PATCH](https://www.belgif.be/specification/rest/api-guide/#document-partial-update)
* constraint violation on [DELETE](https://www.belgif.be/specification/rest/api-guide/#remove-document)
* duplicate key on [POST](https://www.belgif.be/specification/rest/api-guide/#create-resource)

But no standardized Belgif problem type is currently provided. It should probably use the InputValidationProblem model, so additional details can be returned in the issues property.

**Comment by pvdbosch** [comment]
We should probably change to http code 412 for optimistic locking scenarios; there's still open issue #37 for that.
For the others, we'll need to evaluate if single level of problem suffices for most of these use cases vs more complex InputValidationProblem model that allows bundling of multiple issues.

**Comment by jpraet** [comment]
> we'll need to evaluate if single level of problem suffices for most of these use cases vs more complex InputValidationProblem model that allows bundling of multiple issues.

Yes, that relates to the discussion on https://github.com/belgif/rest-guide/issues/114#issuecomment-1482464264.

**Comment by pvdbosch** [comment]
409 Conflict may also be used for PUT operation when a resource can't be updated bc of business rules; similar to the above DELETE scenario.
Do we need a generic problem type for such business-specific cases or should it be an API-specific problem type?

**Comment by pvdbosch** [comment]
I've opened new issue #217 replacing this one, and also including cases like replacedSsin, mergedEnterprise, etcetera. It doesn't include optimistic locking, which is covered by #37.
